### PR TITLE
docs: example for sp-underlay

### PR DESCRIPTION
**Pull request by dev-kml**
## Description
Updating `packages/underlay/README.md` and providing an example of usage for `<sp-underlay>`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/underlay/README.md [hunk 1]
```diff
@@ -22,3 +22,13 @@ When looking to leverage the `Underlay` base class as a type and/or for extensio
...
+
+## Example
+
+```html
+<sp-underlay open></sp-underlay>
+<div class="overlay">
+    <h1>Hello, I'm an overlay!</h1>
+    <p>Enjoy your day...</p>
+</div>
+```
```
Comment (Westbrook) [inline packages/underlay/README.md:34]: This needs just a little bit more as the `<sp-underlay open>` will by default occlude the whole page...

```suggestion
```html
When leveraging an `<sp-underlay>` in conjunction with overlay content, place it as a sibling prior to your overlay content. 

```html
<style>
sp-underlay:not([open]) + sp-dialog {
    display: none;
}
sp-underlay + sp-dialog {
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: 1;
    background: var(--spectrum-dialog-confirm-background-color, var(--spectrum-alias-background-color-default));
}
</style>
<sp-button
    onclick="
        console.log(this.nextElementSibling);
        this.nextElementSibling.open = true;
    "
>Open dialog with underlay element</sp-button>
<sp-underlay></sp-underlay>
<sp-dialog size="small">
    <h1 slot="heading">Hello, I'm an overlay!</h1>
    <p>Enjoy your day...</p>
    <sp-button
        slot="button"
        onclick="
            this.parentElement.previousElementSibling.open = false;
        "
    >Close</sp-button>
</sp-dialog>
```
```

**Review by Westbrook — CHANGES_REQUESTED** [review]
Thanks for getting into this! 🥳 Next you (probably me, again, after not looking at it for a while) will thank you, too. 

https://github.com/adobe/spectrum-web-components/blob/0a53f1572fd8e4fde4a61ad1c7162e7c1281fb77/packages/underlay/src/Underlay.ts#L31

The above should be:
```
@property({ type: Boolean, reflect: true })
```
And then usage will be even nicer for this element and its documentation. :pray:

File: packages/underlay/README.md (hunk 1 above)
Comment (dev-kml) [reply packages/underlay/README.md]: Great suggestion! updated: https://github.com/adobe/spectrum-web-components/pull/1681/commits/aeadd1bb4d8f090183903c980fcf576098a0b800

**Review by Westbrook — APPROVED** [review]
This looks great. Thanks so much!!

**Pull request merged**